kclvm/tools/src/query/override.rs

```diff
@@ -1,50 +1,242 @@
+use std::collections::{HashMap, HashSet};
+
+use anyhow::{anyhow, Result};
+
 use kclvm_ast::walker::MutSelfMutWalker;
 use kclvm_ast::{ast, walk_if_mut};
-use kclvm_parser::parse_expr;
+use kclvm_parser::{parse_expr, parse_file};
+
+use crate::printer::print_ast_module;
+
```

Comment:
Two suggestions:
1. Mark the internal APIs and the external APIs. And the external APIs should be move to a separated API crate, and leave a todo about it.
2. For some helper functions that may be general, consider moving them to a reasonable crate, not in the query crate

Reply:
> Two suggestions:
> 
> 1. Mark the internal APIs and the external APIs. And the external APIs should be move to a separated API crate, and leave a todo about it.
> 2. For some helper functions that may be general, consider moving them to a reasonable crate, not in the query crate

Good suggestions. Reorganized API structures.

Reply:
This practice can be extended to all modules and added to the dev guide.